rebar.config

```diff
@@ -27,8 +27,8 @@
 {escript_emu_args, "%%! +sbtu +A1\n"}.
 %% escript_incl_extra is for internal rebar-private use only.
 %% Do not use outside rebar. Config interface is not stable.
-{escript_incl_extra, [{"relx/priv/templates/*", "_build/default/lib/"},
-                      {"rebar/priv/templates/*", "_build/default/lib/"}]}.
+{escript_incl_extra, [{relx, "priv/templates/*"},
+                      {rebar, "priv/templates/*"}]}.
```

Comment:
I don't think this works as a change. This is moving from a wildcard within _any_ directory into something that becomes specific to OTP applications. I.e. an escript might want to bundle things unrelated to OTP apps, and there is no way to maintain that behaviour with the new configuration method.

Reply:
Good point @ferd. I can generalize the implementation of `escript_incl_extra` to not limit its use to OTP apps. To help steer this work, could you please clarify the following?

- `escript_incl_extra` is currently limited to `rebar3`-only internal use, but will it likely become available for other projects?
- What use cases are intended to be supported by `escript_incl_extra`? For instance:
    1. A directory generated by `rebar3` (e.g., `_build/<profile>/lib/`)
    2. A directory under the project root (`rebar.config`), but not generated by `rebar3` (e.g., `priv/`)
    3. A directory outside of the project root (e.g., `/some/other/dir/`). _If any of these use cases are to be ruled out, I'd guess it's this one._

To support (2) and (3), it seems sufficient to allow static paths (as `escript_incl_extra` accepts today).

To support (1), are there additional use cases beyond _an output directory of an OTP app_?

Reply:
We intended for it to only be used by rebar3, but as with all things, people who needed to bundle priv dirs in escripts started relying on it without us ever having improved the mechanism. So while its usage is discouraged, we're fully aware that changing behaviour will break random apps for people.

The use cases we have it for is just carrying priv/ dirs into apps, but not necessarily all of them. So we just did the quick workaround by adding a filepath with a regex. To carry the relx files, it had to be in _build since we don't have it from the root right away, but someone could use any path whatsoever, whether absolute or relative. It's messy, which is the reason we never really made the feature official.

`{escript_priv_dir, [app | {app, Filter}]}` would possibly be a nicer way to support this, that would partially overlap with the existing feature, but I don't know if it'd be cleaner. 

A whole other approach would be the rebar.config.script trick doing detection of _checkouts and changing the paths accordingly, but that's extremely hackish.

Reply:
Just implemented. Do you think it is better to limit the new include wildcards to the app `priv/` directory (as it is with this PR right now), or to keep it general, despite the `rebar3` use case not needing it?

Also, the PR deviates somewhat from the `app | {app, Filter}` suggestion by only supporting `{app, Filter}`. The rationale here is to limit the different syntaxes which would need to be supported. Plus, `{app, "*"}` is arguably not that much worse to type out if needed. 😉 Of course, if there is a strong preference to retain it, it wouldn't be difficult to do so.